src/routes/safe/components/Transactions/helpers/TxEstimatedFeesDetail/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { ReactElement, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import styled from 'styled-components'
+import { Text, ButtonLink, Accordion, AccordionSummary, AccordionDetails } from '@gnosis.pm/safe-react-components'
+
+import { currentSafe, currentSafeThreshold } from 'src/logic/safe/store/selectors'
+import { getLastTxNonce } from 'src/logic/safe/store/selectors/gatewayTransactions'
+import { TxParameters } from 'src/routes/safe/container/hooks/useTransactionParameters'
+import { ParametersStatus, areEthereumParamsVisible } from '../utils'
+import Bold from 'src/components/layout/Bold'
+import { isMaxFeeParam } from 'src/logic/safe/transactions/gas'
+
+const TxParameterWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`
+
+const AccordionDetailsWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+`
+
+const StyledButtonLink = styled(ButtonLink)`
+  padding-left: 0;
+  margin: 8px 0 0 0;
+
+  > p {
+    margin-left: 0;
+  }
+`
+
+const StyledAccordionSummary = styled(AccordionSummary)`
+  & .MuiAccordionSummary-content {
+    justify-content: space-between;
+  }
+`
+
+type Props = {
+  txParameters: TxParameters
+  gasCost: string
+  onEdit: () => void
+  compact?: boolean
+  parametersStatus?: ParametersStatus
+  isTransactionCreation: boolean
+  isTransactionExecution: boolean
+  isOffChainSignature: boolean
+}
+
+export const TxEstimatedFeesDetail = ({
+  onEdit,
+  txParameters,
+  gasCost,
+  compact = true,
+  parametersStatus,
+  isTransactionCreation,
+  isTransactionExecution,
+  isOffChainSignature,
+}: Props): ReactElement | null => {
+  const { nonce } = useSelector(currentSafe)
+  const threshold = useSelector(currentSafeThreshold) || 1
+  const defaultParameterStatus = isOffChainSignature && threshold > 1 ? 'ETH_HIDDEN' : 'ENABLED'
```

Comment:
I've removed this in the approve modal refactoring, we'll need to merge carefully.